devicetypes/smartthings/philio-multiple-sound-siren.src/philio-multiple-sound-siren.groovy

```diff
@@ -0,0 +1,314 @@
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	log.debug "settings: ${settings.inspect()}, state: ${state.inspect()}"
+
+	def sound = settings.sound ?: state.sound
+	def duration = settings.duration ?: state.duration
+
+	if (sound != state.sound || duration != state.duration) {
+		state.sound = sound
+		state.duration = duration
+		commands << secure(zwave.configurationV1.configurationSet(parameterNumber: 31, size: 1, configurationValue: [durationMap[duration]]))
+	}
+
+	response(commands)
+}
+
+def parse(String description) {
+	log.debug "parse($description)"
+	def result = null
+	if (description.startsWith("Err")) {
+		if (zwInfo?.zw?.contains("s")) {
+			result = createEvent(descriptionText:description, displayed:false)
+		} else {
+			result = createEvent(
+					descriptionText: "This device failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+					eventType: "ALERT",
+					name: "secureInclusion",
+					value: "failed",
+					displayed: true,
+			)
+		}
+	} else {
+		def cmd = zwave.parse(description, [0x98: 1, 0x20: 1, 0x70: 1])
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned ${result?.inspect()}"
+	return result
+}
+
+def zwaveEvent(securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x20: 1, 0x85: 2, 0x70: 1])
+	// log.debug "encapsulated: $encapsulatedCommand"
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(basicv1.BasicReport cmd) {
+	log.debug "rx $cmd"
+	handleDeviceEvent(cmd.value)
+}
+
+def zwaveEvent(sensorbinaryv2.SensorBinaryReport cmd) {
+	log.debug "rx $cmd"
+	def result = []
+
+	if (cmd.sensorType == sensorbinaryv2.SensorBinaryReport.SENSOR_TYPE_TAMPER) {
+		result << createEvent(name: "tamper", value: "detected")
+	} else {
+		result = handleDeviceEvent(cmd.sensorValue)
+	}
+	result
+}
+
+def zwaveEvent(notificationv3.NotificationReport cmd) {
+	def result = []
+
+	log.debug "rx $cmd"
+
+	if (cmd.notificationType == notificationv3.NotificationReport.NOTIFICATION_TYPE_BURGLAR) {
+		if (cmd.event == 3) {
+			result << createEvent(name: "tamper", value: "detected")
+		}
+	} else {
+		log.warn "Unknown cmd.notificationType: ${cmd.notificationType}"
+		result << createEvent(descriptionText: cmd.toString(), isStateChange: false)
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.debug "Unhandled Z-Wave command $cmd"
+	createEvent(displayed: false, descriptionText: "$device.displayName: $cmd")
+}
+
+def handleDeviceEvent(value) {
+	log.debug "handleDeviceEvent $value"
+	def result = [
+		createEvent([name: "switch", value: value ? "on" : "off", displayed: false]),
```

Comment:
value == 0xff?

Reply:
I suppose in this use I could do that. Originally I had a bunch of code in NotificationReport because I thought it would report back with what we sent it, but it doesn't, so I could make this explicit.